distributed/diagnostics/plugin.py

```diff
@@ -256,29 +257,34 @@ class PipInstall(WorkerPlugin):
 
     Parameters
     ----------
-    packages : List[str]
-        A list of strings to place after "pip install" command
-    pip_options : List[str]
-        Additional options to pass to pip.
-    restart : bool, default False
-        Whether or not to restart the worker after pip installing
+    packages
+        A list of packages (with optional versions) to install
+    restart
+        Whether or not to restart the worker after installing the packages
         Only functions if the worker has an attached nanny process
 
-    Examples
+    See Also
     --------
-    >>> from dask.distributed import PipInstall
-    >>> plugin = PipInstall(packages=["scikit-learn"], pip_options=["--upgrade"])
-
-    >>> client.register_worker_plugin(plugin)
+    CondaInstall
+    PipInstall
     """
 
-    name = "pip"
+    name: str
+    packages: list[str]
+    restart: bool
 
-    def __init__(self, packages, pip_options=None, restart=False):
+    def __init__(
+        self,
+        packages: list[str],
+        restart: bool,
+    ):
         self.packages = packages
         self.restart = restart
-        self.pip_options = pip_options or []
-        self.id = f"pip-install-{uuid.uuid4()}"
+        self.name = f"{self.installer}-install-{uuid.uuid4()}"
+
+    @abc.abstractproperty
+    def installer(self) -> str:
+        """Return the name of the installer"""
```

Comment:
This feels overdesigned. You could have achieved the same with

```python
class PackageInstall(WorkerPlugin, abc.ABC):
    # Must set in subclasses
    INSTALLER: ClassVar[str]

class PipInstall(PackageInstall):
    INSTALLER = "pip"
```
This won't give you an error when loading the module, but it will still crash the first time you create an instance.

Reply:
Fair point!